control-plane/agents/src/bin/core/controller/scheduling/resources/mod.rs

```diff
@@ -101,6 +102,25 @@ impl PoolItemLister {
             None => vec![],
         }
     }
+    /// Get a list of pool items to resize the replicas that reside on them.
+    pub(crate) async fn list_for_resize(registry: &Registry, spec: &VolumeSpec) -> Vec<PoolItem> {
+        let pool_ids = registry.specs().volume_replica_pools(&spec.uuid);
+
+        let items = Self::nodes(registry)
```

Comment:
> Rather than listing all nodes, you could just list the ones that that the pools

I could take this also: 
> let node_ids = registry.specs().volume_replica_nodes(&spec.uuid);

But subsequently, fetching node wrappers from registry is `async` call, which will be unclean and perhaps not allowed in the iterator to get required node wrappers.